backend/models/User.js

```diff
@@ -1,19 +1,39 @@
 const mongoose = require("mongoose");
-const User = new mongoose.Schema({
+const bcrypt = require('bcrypt');
+
+const UserSchema = new mongoose.Schema({
   firstName: {
     type: String,
-    required: true,
+    required: [true, "Please enter your first name"],
+    lowercase: true,
   },
   lastName: {
     type: String,
-    required: true,
+    required: [true, "Please enter your last name"],
+    lowercase: true,
   },
   email: {
     type: String,
-    required: true,
+    required: [true, "Please enter your email"],
     unique: true,
-    match: /^\w+([.-]?\w+)*@\w+([.-]?\w+)*(\.\w{2,3})+$/,
+    match: [/^\w+([.-]?\w+)*@\w+([.-]?\w+)*(\.\w{2,3})+$/, "Please enter a valid email"],
+    lowercase: true,
+  },
+  password: {
+    type: String,
+    required: [true, "Please enter your password"],
+    minlength: [8, "Minimum password length is 8 characters"],
   },
 });
 
-module.exports = mongoose.model("user", User);
+UserSchema.pre('save', async function(next) {
+  const salt = await bcrypt.genSalt();
+  this.password = await bcrypt.hash(this.password, salt);
+  next();
+});
```

Comment:
before the save on the user, the password is hashed. `getSalt` isn't passed a number, by default its 10 and that seems like a pretty secure place to be for it.